### PR TITLE
Optimize token family header

### PR DESCRIPTION
**Pull request by tchayen**
Fixes TEAM1-15

## What changed (plus any additional context for devs)

Rewrite of list header component to be fast 🚀. Fixed major performance issues.

## Dev checklist for QA: what to test
- Wallet screen. Try to scroll real fast in the prod version. There could be "blanks". This one should not have them.
- Showcase should still work
- Click on family name should work
- Animation of the ">" on click should work too
- family row icons should not flicker

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why **No, re-adding same functionality. Either checked by existing tests or out of scope to add them now.**
- [x] If you added new files, did you update the CODEOWNERS file? **No, not touching important stuff.**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -375,6 +375,7 @@
     "**/simple-get": "4.0.1",
     "**/simple-plist": "1.3.1",
     "**/plist": "3.0.5",
+    "**/jpeg-js": "0.4.4",
```
Comment (osdnk) [inline package.json:378]: ?

**Review by osdnk — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (terrysahaidak) [reply package.json:378]: @tchayen ?

**Comment by jkadamczyk** [comment]
Looks good, but found some weird scroll issues when fetching more transactions, pasted recordings on slack

File: src/components/token-family/TokenFamilyHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import { ButtonPressAnimation } from '../animations';
+import TokenFamilyHeaderIcon from './TokenFamilyHeaderIcon';
+import { Text } from '@rainbow-me/design-system';
+import { ImgixImage } from '@rainbow-me/images';
+import { ThemeContextProps } from '@rainbow-me/theme';
+
+export const TokenFamilyHeaderAnimationDuration = 200;
+export const TokenFamilyHeaderHeight = 50;
+
+const sx = StyleSheet.create({
```
Comment (jkadamczyk) [inline src/components/token-family/TokenFamilyHeader.tsx:15]: When to use `sx` and when to use `cx`? What's the logic here I've used and seen both 😄

File: src/components/token-family/TokenFamilyHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+}: {
+  childrenAmount?: number;
+  emoji?: string;
+  familyImage?: string;
+  isOpen?: boolean;
+  onPress?: () => void;
+  testID?: string;
+  title: string;
+  theme: ThemeContextProps;
+}) => {
```
Comment (jkadamczyk) [inline src/components/token-family/TokenFamilyHeader.tsx:61]: Could you extract interface to be separate and not inline? IMO it is way cleaner. Let me know what you think

**Comment by jkadamczyk** [comment]
Tried on Pixel 6 and Xiaomi Redmi Note 11

A couple of issies:
* The collection logos in headers flash when collapsing and expanding any of the sections
* On lower end device there is still many blanks, but I guess it is expected and the performance isn't terrible

Xiaomi:
https://streamable.com/v9lhzj

Pixel 6:
https://streamable.com/bi6wkx

File: src/components/token-family/TokenFamilyHeader.tsx (hunk 2 above)
Comment (terrysahaidak) [reply src/components/token-family/TokenFamilyHeader.tsx:15]: there were places in the code where `sx` was already used so i changed `cx` to `sx` everywhere recently, sorry for confusions :D

File: src/components/token-family/TokenFamilyHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      easing: Easing.bezier(0.25, 0.1, 0.25, 1),
+      toValue: isOpen ? 0 : 1,
+      useNativeDriver: true,
+    }).start(() => {
+      aRef.current!.isRunning = false;
+    });
+
+    aRef.current!.isRunning = true;
+
+    if (onPress) {
```
Comment (perunt) [inline src/components/token-family/TokenFamilyHeader.tsx:104]: should we use optional chaining for func too?

**Review by perunt — APPROVED** [review]
I tried it on Redmi 7, which is a low-low end device, and it looks good

File: src/components/token-family/TokenFamilyHeader.tsx (hunk 4 above)
Comment (terrysahaidak) [reply src/components/token-family/TokenFamilyHeader.tsx:104]: we could, but personally, i hate these one-liners

```js
onPress?.()
```

**Comment by BrodyHughes** [comment]
Communicating with @tchayen about the PR in dms 👍

**Review by BrodyHughes — APPROVED** [review]
QA looks good! 

After items had loaded checked:
- homescreen scroll
- profile scroll
- send flow scroll

**Pull request merged**